### DISCUSSION TITLE
Broadcast a message to all applications

### DISCUSSION THREAD

**Discussion by adimyth** [svix/svix-webhooks#583] [Ideas]
Is there a mechanism to broadcast a message to all applications?

Scenario - Suppose we have 3 applications that are subscribed to an event "user.signup". If a new user signs up, then how can we notify all the applications (and endpoints subsequently filtered by event type) at once? Currently, the only method available is to call the create message api for each application one by one. This I believe is a common use case.

**Comment by tasn** [comment]
Hey @adimyth, I think you misunderstood the model a bit.

Under normal circumstances you'd have one application per each of your customers. This application can have multiple endpoints (so can listen on multiple different URLs), and you'd send it to this one application which will send to all of the endpoints.

Since applications are different customers of yours it's extremely rare that you'd be sending all of them the same message.

**Reply by adimyth** [reply]
Hey @tasn I totally understand the hierarchical model. However, I do have a use case where all applications must be broadcasted about an event.

**_Example_** - Consider a scenario where a user management system is responsible for notifying all the dependent applications (emailer, group assigner, bill generator, etc). Now these applications are independent of each other, but need to be notified of an event like "user.signup" to do their own work. In such a case, broadcasting would be helpful

**Reply by tasn** [reply]
The scenario you described is exactly the one that would be handled by different endpoints. You'd have one application for all of these.

An application is a consumer target of webhooks. You'd be sending to this one application and it'll send to all of the relevant endpoints you've described.

**Reply by adimyth** [reply]
Yeah, I understand that. Anyways, it made sense to isolate different apps & create separate webhook application for each actual app.
